### ISSUE TITLE
If sneaking and adjusting player collisionbox below slab thickness, collision still occurs 

### ISSUE THREAD

**Issue by sirrobzeroone** [minetest/minetest#10706]
##### Minetest version
```
MT 5.3
```
Operating system: Windows 10
CPU: i7

##### Summary
If you apply a change to the players collision box size when they press sneak and set the collision box to say 1.5 blocks high the player still won't be able to walk under a space that is 1.5 nodes of empty space. ie the player will "catch" on the slab above.   

##### Steps to reproduce
Easiest way to replicate although this replicable outside the MTG -

to player_api >>api.lua add:
Below line 120 add
```
player:set_properties({collisionbox = {-0.3, 0.0, -0.3, 0.3, 1.45, 0.3}})
player:set_properties({eye_height = 1.37})
```
Replace lines 131 to 140 with the below

```
if controls.LMB or controls.RMB then
					player_set_animation(player, "walk_mine", animation_speed_mod)
				
				else
					player_set_animation(player, "walk", animation_speed_mod)
					player:set_properties({collisionbox = {-0.3, 0.0, -0.3, 0.3, 1.7, 0.3}})
					player:set_properties({eye_height = 1.47})
				end
			elseif controls.LMB or controls.RMB then
				player_set_animation(player, "mine", animation_speed_mod)
			else
				player_set_animation(player, "stand", animation_speed_mod)
				player:set_properties({collisionbox = {-0.3, 0.0, -0.3, 0.3, 1.7, 0.3}})
				player:set_properties({eye_height = 1.47})

```

Setup a small run basically front view 2 pillars of solid nodes on each side and 1 slab in the top slot of the upper two nodes basically creating a gap 1.5 high:
```
 ___ ___ ___
|__|----|__|
|__|    |__|

```
Now attempt to crouch-walk through you will stop approximately half-way through the slab and unable to walk further.

Investigations by Skamiz Kazzarch seem to indicate it relates to the edge snapping sneaking does - full forum thread here:
https://forum.minetest.net/viewtopic.php?t=25784

Recording as a bug but may be deliberate system design to help with performance. So no expectation this can be fixed and it maybe better the behaviour is simply documented in the API somewhere. As Skamiz Kazzarch noted in the above thread the behaviour can actually be used to elicit interesting game behaviour.

**Comment by SmallJoker** [comment]
Duplicate, or very closely related to #1229

**Comment by sirrobzeroone** [comment]
I guess the most disappointing thing is that bummer I didnt find a unique bug :( lol

After reading through the other bug report agree with yourself and paramat dont think its worth fixing, given I already have some workaround lua code to get the behaviour I want. i'd rather the devs work on more meaningful/fun stuff :)

**Comment by SmallJoker** [comment]
Closing as duplicate.

**Comment by sirrobzeroone** [comment]
@SmallJoker - I stuffed up with my initial description of this issue I've updated the first post, after Skamiz Kazzarch very carefully re-explained I dont think the other issue and this one are related. My simple understanding is this:

While sneaking player/charcater collides with an upper collision box which then programmatically removes the node on the ground infront of the player and because we are sneaking and sneaking stops us from falling in holes we are effectively on an edge and stop.

Skamiz Kazzarch has amuch better understanding than me of whats happening his last example in the forum thread really shows it well.
